Source/PINDiskCache.m

```diff
@@ -1116,6 +1254,30 @@ - (void)trimToSizeByDate:(NSUInteger)trimByteCount
     [self trimDiskToSizeByDate:trimByteCount];
 }
 
+- (void)removeExpiredObjects
+{
+    [self lockForWriting];
+
+        NSDate *now = [NSDate date];
+        NSMutableArray<NSString *> *expiredObjectKeys = [NSMutableArray array];
+        [_metadata enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, PINDiskCacheMetadata * _Nonnull obj, BOOL * _Nonnull stop) {
+            NSTimeInterval ageLimit = obj.ageLimit > 0.0 ? obj.ageLimit : self->_ageLimit;
+            NSDate *expirationDate = [obj.date dateByAddingTimeInterval:ageLimit];
+            if ([expirationDate compare:now] == NSOrderedAscending) { // Expiration date has passed
+                [expiredObjectKeys addObject:key];
+            }
+        }];
+
+        [self unlock];
+            for (NSString *key in expiredObjectKeys) {
```

Comment:
Can this for-loop be moved outside the lock? Or the `[self lock]` below should be changed to `[self lockForWriting]`?

Reply:
Yes. Moved this outside the lock.